python/hsfs/feature_group.py

```diff
@@ -143,11 +143,11 @@ def filter(self, f: Union[filter.Filter, filter.Logic]):
         """
         return self.select_all().filter(f)
 
-    def add_tag(self, name: str, value: str = None):
+    def add_tag(self, name: str, value: str):
         """Attach a name/value tag to a feature group.
 
-        A tag can consist of a name only or a name/value pair. Tag names are
-        unique identifiers.
+        A tag consists of a name/value pair. Tag names are unique identifiers.
+        The value of a tag can be any valid json - primitives, arrays or json objects.
```

Comment:
`json - primitives, arrays or json objects.` this is not true as you are explicitly asking for a str. I think it would be better if we accept both `str` and `Dict` here and then check the type of the value.
If the value is `Dict` we convert it to json, if it's string, we just send it to the backend.

Reply:
removed the str constrain, allowing any object that can have `json.dumps` applied to it correctly
adding a ruby test in hopsworks for this...